docs/auth.md

```diff
@@ -0,0 +1,65 @@
+# Kubernetes Native Authentication
+
+Kubernetes native authentication defines a flow where the Armada
+Executor cluster's in-built token handling is used to authenticate
+the Executor requesting jobs from the Armada Server.
+
+## Authentication Flow
+
+1. The Executor requests a temporary Service Account Token from its own cluster's 
+Kubernetes API using TokenRequest
+2. The Executor sends this Token to the Server in an Authorization Header.
+3. The Server decodes the token to read the KID (Kubernetes Key ID).
+4. The Server swaps for the Callback URL of the executor's API server.
+The KID to URL mapping is stored in a prior generated ConfigMap or Secret.
+5. The Server uses the URL to call the Executor Cluster's TokenReview endpoint.
+6. On successful review the Server caches the Token for its lifetime,
+on unsuccessful review the Token is cached for a configuration defined time.
+
+## Setup
+
+### Kid-mapping ConfigMap
+
+The Armada Server must have a ConfigMap in its namespace with the following
+format for entries:
+
+```yaml
+data:
+  "<CLUSTER_KID>": "<EXECUTOR_CLUSTER_CALLBACK_URL>"
+  ...
+```
+
+This ConfigMap may be mounted anywhere on the Server's Pod.
```

Comment:
This seems to be a implementation detail of how Armada stores ConfigMaps.  

Is it required that a configmap is stored as a volume or can someone create a configmap as a k8 object?

Reply:
wdym create a configmap as a k8s object?
The configmap needs to be mounted as this is what allows the pod to read updates without having to rotate the pod.
There are alternatives to how you get configmap info into a pod but they usually do not rotate except with the pod.

Reply:
Didn't know that

Reply:
There are four different ways that you can use a ConfigMap to configure a container inside a Pod:

    Inside a container command and args
    Environment variables for a container
    Add a file in read-only volume, for the application to read
    Write code to run inside the Pod that uses the Kubernetes API to read a ConfigMap

Source: https://kubernetes.io/docs/concepts/configuration/configmap/
As I see it 1 and 2 are clearly fixed after first read, 3 updates when the configmap does (though this is controlled by kubernetes and I don't believe is instant, though fast enough for our uses). 4 also clearly updates with the configmap but is the least abstracted way of doing this so we definitely shouldn't be relying on it.